src/app/policies/views/PolicyDetailView.vue

```diff
@@ -5,21 +5,21 @@
     data-testid="policy-detail-view"
   >
     <DataSource
-      v-slot="{ data: policiesData, isLoading: policiesIsLoading, error: policiesError }: PolicyTypeCollectionSource"
+      v-slot="{ data: policyTypesData, error: policyTypesError }: PolicyTypeCollectionSource"
       :src="`/*/policy-types`"
     >
-      <LoadingBlock v-if="policiesIsLoading" />
-
       <ErrorBlock
-        v-else-if="policiesError"
-        :error="policiesError"
+        v-if="policyTypesError"
+        :error="policyTypesError"
       />
 
-      <EmptyBlock v-else-if="policiesData === undefined || policiesData.policies.length === 0" />
+      <LoadingBlock v-else-if="policyTypesData === undefined" />
+
+      <EmptyBlock v-else-if="policyTypesData.policies.length === 0" />
 
       <template v-else>
         <template
-          v-for="currentPolicyType in [policiesData.policies.find((item) => item.path === route.params.policyPath) ?? policiesData.policies[0]]"
+          v-for="currentPolicyType in [policyTypesData.policies.find((item) => item.path === route.params.policyPath) ?? policyTypesData.policies[0]]"
```

Comment:
It's odd we're using a for when there's always a single element no?

Reply:
Yeah, I’m really not a fan of this style of setting a variable in a template through the mechanism of the `v-for` directive. It’s basically a workaround around the fact that with DataSource, data is always scoped to templates and not available in a Vue SFC script section at all (except via function calls). This also makes it impossible to create computed properties from this data which is what one would use here.

Reply:
Can't you use the `:set` mentioned previously no?

Reply:
Unless I’m out of touch with Vue templating, this is not a thing. You can’t declare variables inside templates and the use of the `v-for` loop here is really a workaround around this.

Reply:
I got it from: https://github.com/vuejs/core/issues/1172#issuecomment-1035939776 it sort of makes sense to me.

Looking further down this hole I also noticed: https://dev.to/loilo/an-approach-to-vuejs-template-variables-5aik which does what we do. 

> It’s basically a workaround around the fact that with DataSource, data is always scoped to templates and not available in a Vue SFC script section at all (except via function calls). This also makes it impossible to create computed properties from this data which is what one would use here.

So it seems that this is workaround around what DataSource was made for: getting the data close to where you actually use it (remember the `<img>` analogy). I think there's a @change callback or equivalent that @johncowen created to still be able to have computed properties if it were absolutely required. To me most of the time computed properties are better avoiding because they are implicit unscoped, global variables.